python/ml4ir/base/model/scoring/Monte_Carlo_Scorer.py

```diff
@@ -0,0 +1,131 @@
+
+import logging
+from logging import Logger
+import traceback
+from typing import Dict, Optional, Union, List
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras.metrics import Metric
+
+from ml4ir.base.config.keys import FeatureTypeKey
+from ml4ir.base.features.feature_config import FeatureConfig
+from ml4ir.base.io.file_io import FileIO
+from ml4ir.base.model.architectures import architecture_factory
+from ml4ir.base.model.losses.loss_base import RelevanceLossBase
+from ml4ir.base.model.scoring.interaction_model import InteractionModel
+from ml4ir.base.model.scoring.scoring_model import RelevanceScorer
+
+
+class MonteCarloScorer(RelevanceScorer):
+    """
+    Monte Carlo class that defines the neural network layers that convert
+    the input features into scores by using monte carlo trials in inference.
```

Comment:
@mohazahran worth explicitly calling out that the masking of the features is not done in this and must be done via model config using either dropout, or one of the mask layers available